Utilities/Files.cs

```diff
@@ -88,39 +107,60 @@ public static void Write(string name, string content)
                     return;
                 }
 
-                // Attempt to write the file
-                try
-                {
-                    File.WriteAllText(name, content, Encoding.UTF8);
-                }
-                catch (ArgumentException ex)
-                {
-                    Logging.Error("Failed to write to " + name, ex);
-                }
-                catch (PathTooLongException ex)
-                {
-                    Logging.Error("Path " + name + " too long", ex);
-                }
-                catch (DirectoryNotFoundException ex)
-                {
-                    Logging.Error("Directory for " + name + " not found", ex);
-                }
-                catch (IOException ex)
-                {
-                    Logging.Error("IO exception for " + name, ex);
-                }
-                catch (UnauthorizedAccessException ex)
-                {
-                    Logging.Error("Not allowed to write to " + name, ex);
-                }
-                catch (NotSupportedException ex)
+                await Task.Run(() =>
                 {
-                    Logging.Error("Not supported writing to " + name, ex);
-                }
-                catch (SecurityException ex)
-                {
-                    Logging.Error("Security exception writing to " + name, ex);
-                }
+                    int attempts = 20;
+                    int ioDelayMs = 25;
+
+                    while (attempts > 0)
+                    {
+                        // Attempt to write the file
+                        try
+                        {
+                            LockManager.GetLock(name, () => File.WriteAllText(name, content, Encoding.UTF8));
+                        }
+                        catch (ArgumentException ex)
+                        {
+                            Logging.Error("Failed to write to " + name, ex);
```

Comment:
Likewise I think these will loop infinitely, as attempts is not decremented or zeroed.

Reply:
We break out of the loop, regardless of whether `attempts` decrements to zero, using `break;` at line 157.